src/components/balances/AssetsTable/index.tsx

```diff
@@ -12,118 +13,258 @@ import useIsGranted from '@/hooks/useIsGranted'
+const headCells = [
+  {
+    id: 'asset',
+    label: 'Asset',
+    width: '60%',
+  },
+  {
+    id: 'balance',
+    label: 'Balance',
+    width: '20%',
+  },
+  {
+    id: 'value',
+    label: 'Value',
+    width: '20%',
+  },
+  {
+    id: 'actions',
+    label: '',
+    width: '20%',
+    sticky: true,
+  },
+]
+
+const AssetsTable = ({
+  showHiddenAssets,
+  setShowHiddenAssets,
+}: {
+  showHiddenAssets: boolean
+  setShowHiddenAssets: (hidden: boolean) => void
+}): ReactElement => {
   const [selectedAsset, setSelectedAsset] = useState<string | undefined>()
   const isGranted = useIsGranted()
+  const [assetsToHide, setAssetsToHide] = useState<string[]>([])
+  const [assetsToUnhide, setAssetsToUnhide] = useState<string[]>([])
+  const hiddenAssets = useHiddenTokens()
+  const { balances } = useBalances(true)
+  const dispatch = useAppDispatch()
+  const chainId = useChainId()
+  const [hidingAsset, setHidingAsset] = useState<string>()
 
-  const shouldHideActions = !isGranted
+  // TODO: move logic into hook
+  const toggleAsset = useCallback(
+    (address: string) => {
+      if (assetsToHide.includes(address)) {
+        assetsToHide.splice(assetsToHide.indexOf(address), 1)
+        setAssetsToHide([...assetsToHide])
+        return
+      }
 
-  const headCells = useMemo(
-    () => [
-      {
-        id: 'asset',
-        label: 'Asset',
-        width: '60%',
-      },
-      {
-        id: 'balance',
-        label: 'Balance',
-        width: '20%',
-      },
-      {
-        id: 'value',
-        label: 'Value',
-        width: '20%',
-      },
-      {
-        id: 'actions',
-        label: '',
-        width: '20%',
-        hide: shouldHideActions,
-        sticky: true,
-      },
-    ],
-    [shouldHideActions],
+      if (assetsToUnhide.includes(address)) {
+        assetsToUnhide.splice(assetsToUnhide.indexOf(address), 1)
+        setAssetsToUnhide([...assetsToUnhide])
+        return
+      }
+
+      const assetIsHidden = hiddenAssets.includes(address)
+      if (!assetIsHidden) {
+        setAssetsToHide([...assetsToHide, address])
+      } else {
+        setAssetsToUnhide([...assetsToUnhide, address])
+      }
+    },
+    [assetsToHide, assetsToUnhide, hiddenAssets],
+  )
+
+  const deselectAll = useCallback(() => {
+    setAssetsToHide([])
+    setAssetsToUnhide([...hiddenAssets])
+  }, [hiddenAssets])
+
+  // Assets are selected if they are either hidden or marked for hiding
+  const isAssetSelected = useCallback(
+    (address: string) =>
+      (hiddenAssets.includes(address) && !assetsToUnhide.includes(address)) || assetsToHide.includes(address),
+    [assetsToHide, assetsToUnhide, hiddenAssets],
+  )
+
+  const visibleAssets = useMemo(
+    () =>
+      showHiddenAssets
+        ? balances.items
+        : balances.items?.filter(
+            (item) => item.tokenInfo.type === TokenType.NATIVE_TOKEN || !hiddenAssets.includes(item.tokenInfo.address),
```

Comment:
Why is the native token filtered out?

Reply:
Currently because I can't imagine people actually want to hide it on purpose and because the native token has no address. 
So we would need some hacky workaround such as "If the Zero address is in the hidden tokens, we do not show the native token"

Reply:
I don't see how it's a reason not to allow the user to do it if they want to. Zero address (0x0000...) is what backend sends us the native token's address IIRC.

Reply:
Okay I'll remove this condition and the disabled flags in the components.
I checked and true: Zero address is returned by backend.